benchmark/arxiv_kaggle/data_generation.py

```diff
@@ -0,0 +1,142 @@
+import json
+import shutil
+from pathlib import Path
+from typing import Annotated, Any
+
+import pandas as pd
+import typer
+from benchmark.utils.time_resolution_binning import (
+    HELP_RESOLUTION,
+    bin_dataframe_wrt_time_resolution,
+    df_to_csv_with_timestamp,
+)
+from modyn.const.types import TimeResolution
+from sklearn.model_selection import train_test_split
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+
+class ArxivKaggleDownloader(Dataset):
+    test_holdout = 0.25
+    fields_to_keep = ["title", "category"]
+
+    def __init__(self, data_dir: Path, data_zip: Path) -> None:
+        self.assert_data_exists(data_zip)
+        self.data_dir = data_dir
+
+    def assert_data_exists(self, data_zip: Path) -> None:
+        assert data_zip.exists(), "Kaggle dataset zip file not found."
+
+    def extract_data(self, data_zip: Path) -> None:
+        shutil.unpack_archive(data_zip, self.data_dir)
+
+    @property
+    def data_json(self) -> Path:
+        return self.data_dir / "arxiv-metadata-oai-snapshot.json"
```

Comment:
Do you assume this is downloaded by the user? It's not  clear to me where this is coming from. Either we need a README similar to CGLM, or implement some download logic

Reply:
i am a bit confused since the class (same for huffpost) is named "Downloader" so I was trying to find the download logic somewhere. Not sure it actually exists

Reply:
The only reliable way that I found for downloading from kaggle is via the [kaggle API](https://github.com/Kaggle/kaggle-api). It also requires user auth, so I just went for the approach that a user of the scripts downloads the dataset file from kaggle via the browser and the just hands the filepath to the script. The `Download` in the classname is, indeed, a bit off.